lib/health_checks/sidekiq.rb

```diff
@@ -1,68 +1,60 @@
 require 'benchmark'
-require 'socket'
+require 'fileutils'
 require 'health_checks/checks/memory_check'
 require 'health_checks/checks/mongoid_check'
 require 'health_checks/checks/redis_check'
 
 module HealthChecks
   module_function
 
-  def sidekiq(config, mongo_databases, redis_configs)
+  def sidekiq(config, mongo_databases, redis_configs, sleep_seconds: 10)
     config.on(:startup) do
-      LivenessServer.new.start(mongo_databases, redis_configs)
+      LivenessThreadCheck.new.start(mongo_databases, redis_configs, sleep_seconds)
     end
   end
 
   private
 
-  class LivenessServer
-    LIVENESS_PORT = 8080
+  STATUS_FILE = '/tmp/sidekiq_ok'
 
-    def start(mongo_databases, redis_configs)
-      Sidekiq::Logging.logger.info "Starting liveness server on #{LIVENESS_PORT}"
+  class LivenessThreadCheck
```

Comment:
Isn't `LivenessCheckThread` a better name? It's confusing when "thread" is in the middle, suggests that you're checking the liveness thread